### PR TITLE
Add TopTabNavigator compontent and add to example screens for testing

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description

- Add TopTabNavigator component
- Add a few more example pages for testing

This PR is in a rough state now bc of rebasing and not being able to build after pulling in from master. Will fix

### How Has This Been Tested?

Manually tested, still needed some review and testing though

### CODE REVIEW

File: src/components/app-navigator/BottomTabNavigator.tsx [hunk 1]
```diff
@@ -80,6 +94,23 @@ type BottomTabNavigatorProps = {
...
+      currentState?.routeNames?.length > 0 &&
+      currentState?.routes?.length > 0
+    ) {
+      routePath.push(currentState.routeNames[currentState.index])
+      currentState = currentState.routes[currentState.index].state
+    }
+
+    console.log({ routePath })
+  }
+
```
Comment (Kyle-Shanks) [inline src/components/app-navigator/BottomTabNavigator.tsx:113]: This was for logging out the path for deep linking research. Will remove after discussion

File: src/screens/track-screen/TrackScreen.tsx [hunk 2]
```diff
@@ -1,24 +1,59 @@
...
 // to get all the relevant params
 type Props = NativeStackScreenProps<BaseStackParamList, 'track'>
 
+const screenHeight = Dimensions.get('window').height
+
+// const InnerFeedScreen = () => {
+//   return <Text>Feed Screen</Text>
+// }
+
+// const BananaScreen = () => {
```
Comment (raymondjacobson) [inline src/screens/track-screen/TrackScreen.tsx:20]: banana!

File: src/components/app-navigator/TopTabNavigator.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+// How much the tab indicator should horizontally stretch
+// while it translates for a nice effect.
+const INDICATOR_STRETCH_FACTOR = 0.35
+
+// This granularity is how many points of the sin function are added to the stretch animation
+// (min value) 1 will not add any points in between and, therefore, there will be no stretching
+// (recommended min value) 2 will be a linear stretch to max width and back
+// Values above 2 will start to conform to a sin function curve of stretching
+// The high the number, the more accurate the line stretch will follow the sin function
```
Comment (sliptype) [inline src/components/app-navigator/TopTabNavigator.tsx:24]:
```suggestion
// The higher the number, the more accurate the line stretch will follow the sin function
```

File: src/components/app-navigator/TopTabNavigator.tsx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    tab: {
+      alignItems: 'center',
+      flex: 1,
+      flexDirection: 'column',
+      height: 48,
+      justifyContent: 'space-evenly'
+    }
+  })
+
+const TestTabBar = ({ state, descriptors, navigation, position }) => {
```
Comment (sliptype) [inline src/components/app-navigator/TopTabNavigator.tsx:67]: Should we name this something else?

File: src/components/app-navigator/TopTabNavigator.tsx [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    getSinAnimationRanges(state.routes.length)
+  )
+
+  return (
+    <View style={styles.tabBarContainer}>
+      {state.routes.map((route, index) => {
+        const { options } = descriptors[route.key]
+        const label = options.tabBarLabel ?? options.title ?? route.name
+        const Icon = options.tabBarIcon?.()
+        console.log({ options, label })
```
Comment (sliptype) [inline src/components/app-navigator/TopTabNavigator.tsx:109]: Can drop this console.log

File: src/components/app-navigator/TopTabNavigator.tsx [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+              testID={options.tabBarTestID}
+            >
+              {Icon && (
+                <Animated.View style={{ opacity }}>
+                  <Icon />
+                </Animated.View>
+              )}
+              <Animated.Text
+                style={{
+                  color: '#858199',
```
Comment (sliptype) [inline src/components/app-navigator/TopTabNavigator.tsx:135]: Does this correspond to a theme color? I think you could define these styles in the StyleSheet and then do something like:

``` 
<Animated.Text style={[styles.text, { opacity }]>
```

File: src/components/app-navigator/TopTabNavigator.tsx [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+              >
+                {label}
+              </Animated.Text>
+            </TouchableOpacity>
+          </View>
+        )
+      })}
+      <Animated.View
+        style={{
+          backgroundColor: '#CC0FE0',
```
Comment (sliptype) [inline src/components/app-navigator/TopTabNavigator.tsx:150]: Same as above ^

File: src/screens/feed-screen/FeedScreen.tsx [hunk 8]
```diff
@@ -1,22 +1,56 @@
...
 
+// import TopTabNavigator from 'app/components/app-navigator/TopTabNavigator'
 import { FeedStackParamList } from 'app/components/app-navigator/types'
 import Button from 'app/components/button'
 
 type Props = NativeStackScreenProps<FeedStackParamList, 'feed'>
 
+const screenHeight = Dimensions.get('window').height
+
+// TODO: Remove TopTabNavigator from this file
```
Comment (sliptype) [inline src/screens/feed-screen/FeedScreen.tsx:14]: This seems done

**Review by sliptype — APPROVED** [review]
Looks good! This is exciting 😄

**Pull request merged**